test/unit/validators.test.ts

```diff
@@ -0,0 +1,86 @@
+import {expect} from "chai";
+import {describe, it} from "mocha";
+import {
+  BigIntUintType,
+  booleanType,
+  ByteVectorType,
+  ContainerType,
+  ContainerLeafNodeStructType,
+  NumberUintType,
+  toHexString,
+  TreeBacked,
+} from "../../src";
+
+describe("Container with BranchNodeStruct", function () {
+  this.timeout(0);
+
+  const Bytes48 = new ByteVectorType({length: 48});
+  const Bytes32 = new ByteVectorType({length: 32});
+  const Uint64 = new BigIntUintType({byteLength: 8});
+  const Number64 = new NumberUintType({byteLength: 8});
+
+  interface Validator {
+    pubkey: Uint8Array;
+    withdrawalCredentials: Uint8Array;
+    effectiveBalance: bigint;
+    slashed: boolean;
+    activationEligibilityEpoch: number;
+    activationEpoch: number;
+    exitEpoch: number;
+    withdrawableEpoch: number;
+  }
+
+  const fields = {
+    pubkey: Bytes48,
+    withdrawalCredentials: Bytes32,
+    effectiveBalance: Uint64,
+    slashed: booleanType,
+    activationEligibilityEpoch: Number64,
+    activationEpoch: Number64,
+    exitEpoch: Number64,
+    withdrawableEpoch: Number64,
+  };
+
+  const ValidatorType = new ContainerType<Validator>({fields});
+  const ValidatorLeafNodeStructType = new ContainerLeafNodeStructType<Validator>({fields});
+
+  const validator: Validator = {
+    pubkey: Buffer.alloc(48, 0xaa),
+    withdrawalCredentials: Buffer.alloc(32, 0xbb),
+    effectiveBalance: BigInt(32e9),
+    slashed: false,
+    activationEligibilityEpoch: 1_000_000,
+    activationEpoch: 2_000_000,
+    exitEpoch: 3_000_000,
+    withdrawableEpoch: 4_000_000,
+  };
+
+  const validatorTb = ValidatorType.createTreeBackedFromStruct(validator);
+  const validatorTbLeafNodeStruct = ValidatorLeafNodeStructType.createTreeBackedFromStruct(validator);
+
+  const ops: Record<string, (treeBacked: TreeBacked<Validator>, type: ContainerType<Validator>) => unknown> = {
```

Comment:
may not need `type` param here.

also, I'd like to add a test to mutate any fields and compare all ops again

Reply:
also, we want to make sure it works in a parent tree, ie create a validator list, modify some properties at some indices and compare the parent root (validator list root) again